tests/Tokenizer/TokensAnalyzerTest.php

```diff
@@ -769,247 +769,236 @@ public function testIsLambdaInvalid()
      */
     public function testIsConstantInvocation($source, array $expected)
     {
-        $tokensAnalyzer = new TokensAnalyzer(Tokens::fromCode($source));
+        $tokens = Tokens::fromCode($source);
+
+        static::assertCount(
+            $tokens->countTokenKind(T_STRING),
+            $expected,
+            'All T_STRING tokens must be tested'
+        );
+
+        $tokensAnalyzer = new TokensAnalyzer($tokens);
 
         foreach ($expected as $index => $expectedValue) {
-            static::assertSame($expectedValue, $tokensAnalyzer->isConstantInvocation($index), 'Token at index '.$index.' should match the expected value.');
+            static::assertSame(
+                $expectedValue,
+                $tokensAnalyzer->isConstantInvocation($index),
+                sprintf('Token at index '.$index.' should match the expected value (%s).', $expectedValue ? 'true' : 'false')
+            );
         }
     }
 
     public function provideIsConstantInvocationCases()
     {
-        return [
```

Comment:
converting array to yielding just for sake of change the approach is making diff longer, review more tricky, and increases the chances for conflict while syncing this change into multiple lines. please avoid in future

Reply:
> converting array to yielding just for sake of change

what about converting array to yielding for avoid code quadruplication?
Look at [`testIsConstantInvocation`](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.19.0/tests/Tokenizer/TokensAnalyzerTest.php#L770-L777), [`testIsConstantInvocation70`](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.19.0/tests/Tokenizer/TokensAnalyzerTest.php#L943-L950) and [`testIsConstantInvocation71`](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.19.0/tests/Tokenizer/TokensAnalyzerTest.php#L972-L979) - exactly the same function content and in this fix, 4th copy - for PHP 8 - needs to be added. Also I want add assertion that all string tokens are checked to make it easier to read.

> making diff longer, review more tricky

it is a burden someone has to bear, cleaning this up wasn't fun either.

> increases the chances for conflict while syncing this change into multiple lines

how come? only changes for this file between 2.19 and 3.0 branches are migration rules.